Bio/Seq.py

```diff
@@ -2184,9 +2184,9 @@ def complement(self):
             d = ambiguous_rna_complement
         else:
             d = ambiguous_dna_complement
-        c = dict([(x.lower(), y.lower()) for x, y in d.items()])
-        d.update(c)
-        self.data = [d[c] for c in self.data]
+        mixed = d.copy()  # We're going to edit this to be mixed case!
+        mixed.update((x.lower(), y.lower()) for x, y in d.items())
+        self.data = [mixed[_] for _ in self.data]
```

Comment:
@MarkusPiotrowski you might be a good person to double check this change?

Reply:
@MarkusPiotrowski this was my commit comment for context:

Fix side effect on ambiguous_dna_complement etc

Spotted while reviewing flake8 F812, reuse of variables in a list comprension. Consider this example prior to the fix:

```python
>>> from Bio.Seq import MutableSeq
>>> from Bio.Data.IUPACData import ambiguous_dna_complement
>>> ambiguous_dna_complement
{'A': 'T', 'C': 'G', 'B': 'V', 'D': 'H', 'G': 'C', 'H': 'D', 'K': 'M', 'M': 'K', 'N': 'N', 'S': 'S', 'R': 'Y', 'T': 'A', 'W': 'W', 'V': 'B', 'Y': 'R', 'X': 'X'}
>>> MutableSeq('acgtACGT').complement()
>>> ambiguous_dna_complement
{'A': 'T', 'C': 'G', 'B': 'V', 'D': 'H', 'G': 'C', 'H': 'D', 'K': 'M', 'M': 'K', 'N': 'N', 'S': 'S', 'R': 'Y', 'T': 'A', 'W': 'W', 'V': 'B', 'Y': 'R', 'X': 'X', 'a': 't', 'c': 'g', 'b': 'v', 'd': 'h', 'g': 'c', 'h': 'd', 'k': 'm', 'm': 'k', 'n': 'n', 's': 's', 'r': 'y', 't': 'a', 'w': 'w', 'v': 'b', 'y': 'r', 'x': 'x'}
>>> quit()
```

Using the MutableSeq object's (in-place) complement method had the unexpected side effect of making the complement mapping dictionary mixed case.

Reply:
Good catch, looks good for me. I was shortly confused by the ```_``` variable, it reminds me of Perl's special variables.

Reply:
Me too - Biopython historically has not used ``_`` as a dummy loop variable, but it seems to be a standard practice and is clear and concise once you are used to it.